lightning/src/util/time.rs

```diff
@@ -58,6 +58,48 @@ impl Sub<Duration> for Eternity {
 	}
 }
 
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+#[cfg(not(feature = "no-std"))]
+pub struct MonotonicTime(std::time::Instant);
+#[cfg(not(feature = "no-std"))]
+static SHIFT: Duration = Duration::from_secs(10 * 365 * 24 * 60 * 60); // 10 years.
+
+#[cfg(not(feature = "no-std"))]
+impl Time for MonotonicTime {
+	fn now() -> Self {
+		let instant = std::time::Instant::now().checked_add(SHIFT).expect("Overflow on MonotonicTime instantiation");
+		Self(instant)
+	}
+
+	fn duration_since(&self, earlier: Self) -> Duration {
+		// On rust prior to 1.60 `Instant::duration_since` will panic if time goes backwards.
+		// However, we support rust versions prior to 1.60 and some users appear to have "monotonic
+		// clocks" that go backwards in practice (likely relatively ancient kernels/etc). Thus, we
+		// manually check for time going backwards here and return a duration of zero in that case.
+		let now = Self::now();
+		if now.0 > earlier.0 { now.0 - earlier.0 } else { Duration::from_secs(0) }
+	}
+
+	fn duration_since_epoch() -> Duration {
+		use std::time::SystemTime;
+		SystemTime::now().duration_since(SystemTime::UNIX_EPOCH).unwrap()
+	}
+
+	fn elapsed(&self) -> Duration {
+		self.0.elapsed()
```

Comment:
Won't this result in an underflow? `Instant::elapsed` uses `Instant::now() - *self`.

Also, note that this method is used in `Writeable for ChannelLiquidity<T>`.

Reply:
lol, yeah, I confused myself. The subtraction occurs in `Instant::elapsed`, but the test that I wrote wasn't catching the failure.